lib/rubocop/cop/sorbet/mutable_constant_sorbet_aware_behaviour.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require "rubocop/cop/style/mutable_constant"
+
+module RuboCop
+  module Cop
+    module Sorbet
+      module MutableConstantSorbetAwareBehaviour
+        def self.prepended(base)
+          base.def_node_matcher(:t_let, <<~PATTERN)
+            (send (const nil? :T) :let $_constant _type)
+          PATTERN
+        end
+
+        def on_assignment(value)
+          t_let(value) do |constant|
+            value = constant
+          end
+
+          super(value)
+        end
+      end
+    end
+  end
+end
+
+RuboCop::Cop::Style::MutableConstant.prepend(
+  RuboCop::Cop::Sorbet::MutableConstantSorbetAwareBehaviour
+)
```

Comment:
Can we move this inside `RuboCop::Cop::Sorbet` so it fits on one line?

Reply:
That's how I had it first, but it felt like the fact that we are modifying the behaviour of a class that we are NOT defining was being lost inside that context, since there are many other things going on in there.

The explicit prepend on its own felt way better to me.